tests/MswWhenTest.test.js

```diff
@@ -100,6 +100,44 @@ test("mocks response including json body data", async () => {
   expect(json).toStrictEqual({ response: "success" });
 });
 
+test("mocks response including json array body data", async () => {
+  when(post("https://some.url")).thenReturnFor(
+    request(withBody([{ "some-key": "some value" }])),
+    ok({ response: "success" })
+  );
+
+  const headers = { "content-Type": "application/json" };
+  const body = JSON.stringify([{ "some-key": "some value" }]);
+  const response = await httpRequest("https://some.url", {
+    method: "POST",
+    headers,
+    body,
+  });
+  const json = await response.json();
+
+  expect(response.status).toBe(200);
+  expect(json).toStrictEqual({ response: "success" });
+});
+
+test("mocks response including json body with embedded array data", async () => {
+  when(post("https://some.url")).thenReturnFor(
+    request(withBody({ "some-key": ["some value"] })),
+    ok({ response: "success" })
+  );
+
+  const headers = { "content-Type": "application/json" };
+  const body = JSON.stringify([{ "some-key": ["some value"] }]);
```

Comment:
Im not sure what was intended here, either line 124 should be 
```
request(withBody([{ "some-key": ["some value"] }])),
``` 
(outer array added)

or line 129 should be 
```
const body = JSON.stringify({ "some-key": ["some value"] });
```
(outer array removed)

Reply:
Good spot! I was going for the second one to check a specific use case. I think this works with the update.